core/tee/tee_fs_common.c

```diff
@@ -556,8 +557,13 @@ static struct tee_fs_file_meta *tee_fs_open_meta_file(
 
 	get_meta_filepath(file, version, meta_path);
 	res = read_meta_file(meta_path, meta);
-	if (res < 0)
+	if (res < 0) {
+		if (res == (int)TEE_ERROR_CORRUPT_OBJECT) {
+			meta->info.length = TEE_ERROR_CORRUPT_OBJECT;
```

Comment:
Would be cleaner if `tee_fs_open_meta_file` also output a TEE_Result as part of its argument, and return NULL here.